### ISSUE TITLE
Allow all file types to be uploaded to File content type

### ISSUE THREAD

**Issue by VictoriaSunNIH** [NCIOCPL/cgov-digital-platform#1521]
Find a way to whitelist all file content types for the File content type.

**Comment by BrendanBeltzNIH** [comment]
This is an active issue https://www.drupal.org/project/drupal/issues/997900. (The guy who caused it even expresses guilt).

Comments indicate that the current patch is no longer cleanly applicable to recent versions of Drupal 8.

Another approach suggested on a cursory search:
https://drupal.stackexchange.com/questions/261810/how-to-allow-any-file-extension-or-use-a-blacklist-instead-of-a-whitelist-for

Still needs to be discussed and explored.

**Comment by bryanpizzillo** [comment]
After some fits and starts digging into the mess that is the file upload handling module, I found a solution.
* I added a form alter for the add & editor forms.
* I set `$form['field_media_file']['widget'][0]['#upload_validators']['file_validate_extensions'] = [NULL];`. That is the right magic to bypass the extension check. 
* I also added in a stub function for handling our own validation, such as blacklisting files. This should be easy enough for someone to add to. I have tested it is wired up and working.

One thing to note, all of the file handling logic is currently being rewritten, the beginning of which shipped (late) in 8.7. So all of the file upload functions list TODOs to clean up once the file handler service is in place (which it is).
